meos/src/point/tpoint.c

```diff
@@ -67,6 +67,32 @@ geo_copy(const GSERIALIZED *g)
  * Conversion functions
  *****************************************************************************/
 
+/**
+ * @ingroup meos_internal_temporal_accessor
+ * @brief Return the last argument initialized with the spatiotemporal box of a
+ * temporal point
+ * @param[in] temp Temporal point
+ * @param[out] box Temporal box
+ */
+void
+tpoint_set_stbox(const Temporal *temp, STBox *box)
+{
+  assert(temp); assert(box); assert(tgeo_type(temp->temptype));
+  assert(temptype_subtype(temp->subtype));
+  switch (temp->subtype)
+  {
+    case TINSTANT:
+      tpointinst_set_stbox((TInstant *) temp, box);
+      break;
+    case TSEQUENCE:
+      tpointseq_set_stbox((TSequence *) temp, box);
+      break;
+    default: /* TSEQUENCESET */
+      tpointseqset_set_stbox((TSequenceSet *) temp, box);
+  }
+  return;
+}
```

Comment:
Note that the `tpoint` GiST and SP-GiST functions are also used by the `tnpoint` type.
This means that this function can be called by `tnpoint` values, which currently crashes for `TINSTANT` subtypes, because `tpointinst_set_stbox` cannot handle `tnpoint` values.

This function should thus probably be renamed to `tspatial_set_stbox` to denote the fact that it can be used by all types with an `STBox`. The `tspatial_type` function can then be used in the assert instead of `tgeo_type`. (Same with `tpointinst_set_stbox`, `tpointseq_set_stbox` and `tpointseqset_set_stbox`)